### PR TITLE
refactor(cherow) numeric literal scanning

### PR DESCRIPTION
**Pull request by KFlash**
This is the numeric literals scanning. Feedbacks and ideas for simplification are appreciated

It feels too complex.

### CODE REVIEW

File: packages/cherow/src/lexer/numeric.ts [hunk 1]
```diff
@@ -69,18 +69,16 @@ export function scanNumeric(parser: Parser, context: Context): Token {
...
 
   const end = parser.index;
 
-  let bigInt = false;
-
   if (parser.source.charCodeAt(parser.index) === Chars.LowerN) {
-      if (isFloat) report(parser, Errors.Unexpected);
-      bigInt = true;
+      if (state & NumericState.IsFloat) report(parser, Errors.Unexpected);
+      state = state |= NumericState.IsBigInt;
```
Comment (g-plane) [inline packages/cherow/src/lexer/numeric.ts]: Should it be `state |= NumericState.IsBigInt`?

Comment (KFlash) [reply packages/cherow/src/lexer/numeric.ts]: Maybe it will work?

Comment (g-plane) [reply packages/cherow/src/lexer/numeric.ts]: Run tests to validate it.

Comment (KFlash) [reply packages/cherow/src/lexer/numeric.ts]: There exist tests for this, and the tests pass. Except for in the CLI.

**Pull request merged**